### ISSUE TITLE
SetKeyboardFocusHere(-1) not working with InputTextMultiline

### ISSUE THREAD

**Issue by actondev** [ocornut/imgui#4761]
Version: v1.85
Branch: master

When doing something like
```c++
InputTextMultiline(..);
// checking for right click
if(IsItemHovered() && IsItemClicked(1)) {
SetKeyboardFocusHere(-1);
}
```
Nothing happens.
However, when using `InputText` instead of `InputTextMultiline`, it works as expected!

Another note: putting the `SetKeyboardFocusHere(0)` before the `InputTextMultiline`does have effect (not useful but nonetheless).

Some context: I noticed that only with left mouse-click does an inputText gets focus & wanted to have it also happening with the right click. But as I saw in the implementation, the `MouseClicked[0]` etc checks are hardcoded, taking into account only the left click.

**Comment by ocornut** [comment]
Reopening as this commit broke tabbing through `InputTextMultiline()` multiline.

**Comment by ocornut** [comment]
Pushed a fix c5db276.
It's not great, will simplify and tidy up the mess as we attempt to remove the child window from `InputTextMultiline().`.

**Comment by Skulaurun** [comment]
Same issue on `dear imgui, v1.90.4` master branch. I have followed the tutorial for Debug Console in `imgui_demo.cpp`, and after I changed `InputText` to `InputTextMultiline`, the `ImGui::SetKeyboardFocusHere(-1)` stopped restoring focus after pressing Enter.
```
ImGuiInputTextFlags inputFlags = ImGuiInputTextFlags_EnterReturnsTrue
    | ImGuiInputTextFlags_EscapeClearsAll
    | ImGuiInputTextFlags_AllowTabInput
    | ImGuiInputTextFlags_CtrlEnterForNewLine;
if (ImGui::InputTextMultiline("Lingo v0.1", inputBuffer.data(), inputBuffer.size(), {}, inputFlags)) {
    std::size_t dataLength = strlen(inputBuffer.data());
    execute(inputBuffer.substr(0, dataLength));
    std::memset(inputBuffer.data(), '\0', dataLength);
    ImGui::SetKeyboardFocusHere(-1);
}
```
Is there a workaround to keep the multiline text input focused after pressing enter?

**Comment by ocornut** [comment]
Please open a new issue for increased visibility.

**Comment by ocornut** [comment]
Fixed again 521f84a3a + extra test https://github.com/ocornut/imgui_test_engine/commit/633988ec575ceca4e3f75660fa390b521469573e
